tests/functional/registration/projects/flow_dsl/tests/test_issue196.py

```diff
@@ -0,0 +1,116 @@
+from textx import metamodel_for_language
+from textx.exceptions import TextXError
+import os.path
+import pytest
+
+
+current_dir = os.path.dirname(__file__)
+
+
+def test_issue196_errors_in_scope_provider_and_obj_processor():
+    mmF = metamodel_for_language('flow-dsl')
+
+    # print("-----------------------------------1---")
+    # print(metamodel_for_language('flow-dsl').
+    #      _tx_model_repository.all_models.filename_to_model.keys())
+
+    cached_model_count = len(
+        metamodel_for_language('flow-dsl').
+        _tx_model_repository.all_models.filename_to_model.keys())
+
+    with pytest.raises(TextXError,
+                       match=r'.*types must be lowercase.*'):
+        mmF.model_from_file(os.path.join(current_dir,
+                                         'models',
+                                         'MODEL_WITH_IMPORT_ERROR.eflow'))
+
+    # print("-----------------------------------2---")
+    # print(metamodel_for_language('flow-dsl').
+    #      _tx_model_repository.all_models.filename_to_model.keys())
+
+    # error while reading, no file cached!
+    assert cached_model_count == len(
+        metamodel_for_language('flow-dsl').
+        _tx_model_repository.all_models.filename_to_model.keys())
```

Comment:
`keys()` is superfluous. BTW, the comment says `no file cached`. Should we check that `len` returns `0`?
There are more instances of this bellow.

Reply:
Removed `keys()`.
Added comment to clarify that the count is checked to be unchanged (and not 0).